src/storage/compactor/src/server.rs

```diff
@@ -272,3 +279,261 @@ pub async fn compactor_serve(
 
     (join_handle, observer_join_handle, shutdown_send)
 }
+
+pub async fn shared_compactor_serve(
+    listen_addr: SocketAddr,
+    advertise_addr: HostAddr,
+    opts: CompactorOpts,
+) -> (JoinHandle<()>, Sender<()>) {
+    type CompactorMemoryCollector = HummockMemoryCollector;
+
+    let config = load_config(&opts.config_path, &opts);
+    info!("Starting compactor node",);
+    info!("> config: {:?}", config);
+    info!(
+        "> debug assertions: {}",
+        if cfg!(debug_assertions) { "on" } else { "off" }
+    );
+    info!("> version: {} ({})", RW_VERSION, GIT_SHA);
+
+    // In dedicated compaction mode, these parameters are load from storage opt,
+    // and in shared compaction mode, these parameters should be defined via cloud infra.
+    let parallel_compact_size_mb: u32 = 0;
+    let worker_num: u32 = 0;
+    let max_sub_compaction: u32 = 0;
+    let block_size_kb: u32 = 0;
+    let object_store_recv_buffer_size: usize = 0;
+    let sstable_size_mb: u32 = 0;
+    let bloom_false_positive: f64 = 0.0;
+    let compactor_max_sst_size: u64 = 0;
+    let compact_iter_recreate_timeout_ms: u64 = 0;
+
+    let meta_cache_capacity_mb: usize = 0;
+
+    // in shared compaction mode, these object stoage related parameters should be defined via cloud
```

Comment:
These object storage level parameters should be passed by proxy via cloud means.